### PR TITLE
Support multiple notifier for extraction diff

### PR DESCRIPTION
**Pull request by aurambaj**
this is prep work to support multiple notifiers to send the extraction diff statistics.

### CODE REVIEW

File: cli/src/main/java/com/box/l10n/mojito/cli/command/ExtractionDiffNotificationCommand.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+      extractionDiffStatistics =
+          extractionDiffService.computeExtractionDiffStatistics(extractionDiffPaths);
+    } catch (MissingExtractionDirectoryExcpetion missingExtractionDirectoryExcpetion) {
+      throw new CommandException(
+          "Can't compute extraction diff statistics", missingExtractionDirectoryExcpetion);
+    }
+
+    if (shouldSendNotification(extractionDiffStatistics)) {
+      for (ExtractionDiffNotifier extractionDiffNotifier : extractionDiffNotifiers) {
+        String message = extractionDiffNotifier.sendDiffStatistics(extractionDiffStatistics);
```
Comment (aurambaj) [inline cli/src/main/java/com/box/l10n/mojito/cli/command/ExtractionDiffNotificationCommand.java:193]: need to figure this out

File: cli/src/test/java/com/box/l10n/mojito/cli/command/extractiondiffnotifier/ExtractionDiffNotifierGithubTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        new ExtractionDiffNotifierGithub(
+            new ExtractionDiffNotifierMessageBuilder("{baseMessage}"),
+            mockGithubClient,
+            repository,
+            prNumber);
+
+    final String msg =
+        extractionDiffNotifierGithub.sendDiffStatistics(ExtractionDiffStatistics.builder().build());
+
+    assertThat(msg).isEqualTo("ℹ️ 0 strings removed and 0 strings added (from 0 to 0)");
```
Comment (aurambaj) [inline cli/src/test/java/com/box/l10n/mojito/cli/command/extractiondiffnotifier/ExtractionDiffNotifierGithubTest.java:31]: @maallen i see you've introduced a GithubIcons for the other command. When I tried at least the info icon, it was working with the unicode characters. is it the other ones that require it?

Comment (maallen) [reply cli/src/test/java/com/box/l10n/mojito/cli/command/extractiondiffnotifier/ExtractionDiffNotifierGithubTest.java:31]: I was using the Github markdown syntax for the icons but it appears that it also supports the unicode characters too.

ℹ️ ⚠️ 🛑

Comment (aurambaj) [reply cli/src/test/java/com/box/l10n/mojito/cli/command/extractiondiffnotifier/ExtractionDiffNotifierGithubTest.java:31]: ok going for the unicode chars until proven problematic then

File: cli/src/main/java/com/box/l10n/mojito/cli/command/ExtractionDiffNotificationCommand.java (hunk 1 above)
Comment (aurambaj) [reply cli/src/main/java/com/box/l10n/mojito/cli/command/ExtractionDiffNotificationCommand.java:204]: Adding the parameter for now --console-message-template and only log once

**Pull request merged**